entrypoint.sh

```diff
@@ -48,17 +49,22 @@ pip3 install -r requirements.txt
 # crudini is not available as an Alpine pkg, so we install via pip.
 pip3 install crudini
 
-# Uses GitHub pages API
-# https://docs.github.com/en/rest/pages
-auth_header="Authorization: Bearer ${github_token}"
-accept_header="Accept: application/vnd.github+json"
-version_header="X-GitHub-Api-Version: 2022-11-28"
-page_api_url="https://api.github.com/repos/${GITHUB_REPOSITORY}/pages"
-
-print_site_url_code="import sys, json; print(json.load(sys.stdin)['html_url'])"
-# Get the GitHub pages URL
-github_pages_url_with_trailing_slash=$(curl -L -H "$accept_header" -H "$auth_header" -H "$version_header" "$page_api_url" | python3 -c "${print_site_url_code}")
-github_pages_url=${github_pages_url_with_trailing_slash%/}
+if [ -z "$site_url" ]; then
+    echo "Setting up site URL from GitHub pages API"
+    # Uses GitHub pages API
+    # https://docs.github.com/en/rest/pages
+    auth_header="Authorization: Bearer ${github_token}"
+    accept_header="Accept: application/vnd.github+json"
+    version_header="X-GitHub-Api-Version: 2022-11-28"
+    page_api_url="https://api.github.com/repos/${GITHUB_REPOSITORY}/pages"
+
+    print_site_url_code="import sys, json; print(json.load(sys.stdin)['html_url'])"
+    # Get the GitHub pages URL
+    github_pages_url_with_trailing_slash=$(curl -L -H "$accept_header" -H "$auth_header" -H "$version_header" "$page_api_url" | python3 -c "${print_site_url_code}")
```

Comment:
If the GH Pages is configured with a custom domain name, this would have returned such URL, wouldn't it?

Reply:
Yes. But my use case is I don't won't to host the archive via GitHub pages, so I don't turn on the GitHub pages feature at all. If we use the original script, it will fail exit with this `curl` call return 404.

Reply:
I see. I think that needs additional documentation, i.e. that the branch without explicit URL covers custom URL, but only when the archive is hosted on GH Pages.

Reply:
OK. @rht do you mean you generally apporve this change, but would like to see more docs on the input argument?

Reply:
Yes, as a comment in entrypoint.sh.

Reply:
Because otherwise people would assume that the explicit URL is needed even for GH pages with custom domain name. Actually, in addition to commenting on entrypoint.sh, it also needs to be documented in the README.md. Otherwise, the option is not visible.

Reply:
@rht I pushed a new commit to add docs. PTAL.